code-challenges.test.js

```diff
@@ -17,35 +17,166 @@
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
 
 // b) Create the function that makes the test pass.
+//PseudoCode
+//Input: colors1
+//Output:["yellow", "blue", "pink", "green"] in varying orders
+//Process: create function called newColors with a parameter of any array, then create a new variable to store the new value, after removing the first value from the original array. Then, iterate through the new array with a for loop. Create a new variable to store the randomized shuffle, called newShuff followed by math.random and math.floor to shuffle the entirety of the new array [i]
+//**having issues with the terminal running the function/it just takes away the file path, without giving an outcome
+
+const newColors = (array) => {
+  const shuffleColors = colors1.shift();
+for (let i = colors1.length -1; i>0; i++) {
+  const newShuf = Math.floor(Math.random() * (i+1));
+}
+}
+console.logt(newColors(colors1))
+
+
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+// const votes1 = { upVotes: 13, downVotes: 2 }
+// // Expected output: 11
+// const votes2 = { upVotes: 2, downVotes: 33 }
+// // Expected output: -31
+
+// describe("newVotes", () => {
+//   it("create a function that takes an object that contains up & down votes and returns the total tally", () => {
+//     expect(newVotes(votes1)).toEqual("11")
+//     expect(newVotes(votes2)).toEqual("-31")
+//   })
+// })
+
+//RED- Good failure
+// FAIL  ./code-challenges.test.js
+// newVotes
+//   ✕ create a function that takes an object that contains up & down votes and returns the total tally (1 ms)
+
+// ● newVotes › create a function that takes an object that contains up & down votes and returns the total tally
+
+//   ReferenceError: newVotes is not defined
+
+//     75 | describe("newVotes", () => {
+//     76 |   it("create a function that takes an object that contains up & down votes and returns the total tally", () => {
+//   > 77 |     expect(newVotes(votes1)).toEqual("11")
+//        |     ^
+//     78 |     expect(newVotes(votes2)).toEqual("-31")
+//     79 |   })
+//     80 | })
+
+//     at Object.expect (code-challenges.test.js:77:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.162 s, estimated 1 s
 
 // b) Create the function that makes the test pass.
+//PseudoCode
+//Input: votes1 and votes2
+//Output:11, -31
+//Process: create a function called newVotes that takes a number as a parameter. Then create a new variable to upVotes and downVotes that stores the difference between them. Apply a math equation to find the difference(?). Then console log newVotes with the appropriate pre set variable.
```

Comment:
Thought process is there again, but executing it is the blocker! Also the parameter will represent an object, not a number.